rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -144,24 +167,22 @@ def __next__(self):
             self.labels_padded = torch.LongTensor([row + [0] * (max_cols1 - len(row)) for batch in self.labels_padded for row in batch])
             self.labels_padded = self.labels_padded.view(-1, max_rows1, max_cols1)
 
-            if self.tensor_dtype == types.FLOAT:
-                return torch.from_numpy(self.out),self.bb_padded, self.labels_padded
-            elif self.tensor_dtype == types.FLOAT16:
-                return torch.from_numpy(self.out.astype(np.float16)),self.bb_padded, self.labels_padded
+            return self.out,self.bb_padded, self.labels_padded
 
         else:
             if(self.loader._oneHotEncoding == True):
                 self.loader.GetOneHotEncodedLabels(self.labels)
                 self.labels_tensor = torch.from_numpy(self.labels).type(torch.LongTensor)
                 self.labels_tensor = self.labels_tensor.view(-1, self.bs, self.loader._numOfClasses)
             else:
+                if self.display:
+                    for i in range(self.bs):
+                        img = (self.out)
+                        draw_patches(img[i], i, 0)
                 self.loader.getImageLabels(self.labels)
```

Comment:
please allocate labels as GPU tensor for device and copy to device memory directly in getImageLables rocal c++ call

Reply:
Done & tested the example file